core/src/main/java/org/fao/geonet/kernel/datamanager/base/BaseMetadataManager.java

```diff
@@ -935,7 +935,17 @@ public Element updateFixedInfo(String schema, Optional<Integer> metadataId, Stri
             env.addContent(schemaLoc);
 
             if (updateDatestamp == UpdateDatestamp.YES) {
-                env.addContent(new Element("changeDate").setText(new ISODate().toString()));
+                String changeDate = new ISODate().toString();
+                String createDate = "";
+                if (metadataId.isPresent()) {
+                    java.util.Optional<Metadata> record = metadataRepository.findById(metadataId.get());
+                    if (record.isPresent()) {
+                        changeDate = record.get().getDataInfo().getChangeDate().getDateAndTime();
```

Comment:
update-fixed-info is applied before the metadata is saved in the database, no? I think this code will retrieve the previous change date?

Reply:
Metadata object is modified before so this is fine.
But it does not work on newly created record because the record is not yet in the database. Not sure how can this be solved without adding the creation date as method parameter?